ngrinder-controller/src/main/java/org/ngrinder/agent/controller/AgentManagerController.java

```diff
@@ -65,26 +67,28 @@
 
 	/**
 	 * Get the agents.
-	 *
-	 * @param region the region to search. If null, it returns all the attached
-	 *               agents.
-	 * @param model  model
-	 * @return agent/list
 	 */
 	@RequestMapping({"", "/", "/list"})
-	public String getAll(@RequestParam(value = "region", required = false) final String region, ModelMap model) {
+	@PreAuthorize("hasAnyRole('A', 'S', 'U')")
+	public String getAll(final User user, @RequestParam(value = "region", required = false) final String region, ModelMap model) {
+		final Collection<? extends GrantedAuthority> authorities = getCurrentAuthorities();
 		List<AgentInfo> agents = agentManagerService.getAllVisible();
```

Comment:
What about taking agents as ```Collection<AgentInfo> agents``` from here.
If so, we can just re-use agents variable in the following codes.